### ISSUE TITLE
Unable to debug asp.net core api (.net core 2.1 / VS 2017)

### ISSUE THREAD

**Issue by xs2bharat** [aspnet/Home#3210]
I am unable to debug sample asp.net core web api application created through https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-2.1. 

I am using windows 10 / iis express / .net core 2.1 / latest vs 2017

Getting error: An error occurred attempting to determine the process id of dotnet.exe which is hosting your application. A task was canceled.

There was warning in event log: The directory specified for caching compressed content C:\inetpub\temp\IIS Temporary Compressed Files\xxxxx AppPool is invalid.  Static compression is being disabled.

Even though above path is wrong as i am using iis express, After creating above path, nothing get logged in event log but still getting error.

**Comment by Tratcher** [comment]
Please report VS debugging issues through the VS feedback tool.

**Comment by Tratcher** [comment]
Can you run without debugging (CTL+F5)?

**Comment by xs2bharat** [comment]
Yeah I can

**Comment by Tratcher** [comment]
I've primarily seen this for sites that have enabled client certificates or windows auth. Have you enabled any non-default IIS Express settings?

**Comment by xs2bharat** [comment]
Yup initially i was thinking same, but everything is default. I can attach to dotnet.exe after running thru ctrl + f5 and debug it then,

**Comment by RodrigoFCampos** [comment]
I am using windows 10 / iis / .net core 2.1 / latest vs 2017 and I have the some problem and yes, the only way to debug it is to attach dotnet.exe after running ctrl + f5.

**Comment by Eilon** [comment]
Glad you got it working! Closing this issue.

**Comment by aesser** [comment]
I had a similar issue when running multiple (two) startup projects. Switching to start the .net core project alone and so going back to run both helped in my case.

**Comment by Aspro1** [comment]
I encountered this error when trying to launch a downloaded project in which the target framework version was different from the version installed in my environment. I opened the “.csproj” file in a text editor, corrected the version in the “TargetFramework” node and saved the modified file. In order to be sure to make the changes take effect, I closed and reopened the Visual Studio solution containing the project. Then the project started correctly.
